src/main/java/org/opentripplanner/routing/trippattern/TripTimes.java

```diff
@@ -474,6 +478,66 @@ public boolean tripAcceptable(final State state0, final int stopIndex) {
         return true;
     }
 
+    /**
+     * Returns true if the current state chain contains a banned trip, or banned trip/stop index or banned trip sequence
+     */
+    public boolean tripOrTripSequenceIsBanned(State state0, final int stopIndex) {
+        final RoutingRequest options = state0.getOptions();
+
+        // check if an exact trip ID is banned
+        final BannedStopSet banned = options.bannedTrips.get(trip.getId());
+        if (banned != null && banned.contains(stopIndex)) {
+            return true;
+        }
+
+        // check if a trip ID sequence is banned, but first return false if there are not banned sequences
+        if (options.bannedTripSequences.size() == 0) {
```

Comment:
Could replace with: options.bannedTripSequences.isEmpty()

Reply:
refactored in 6adec27faa2e744f94cd91e4ae4d337a5c25c7c7